src/test/serialize_tests.cpp

```diff
@@ -11,11 +11,38 @@
 #include <stdint.h>
 
 #include <boost/test/unit_test.hpp>
+#include <boost/optional.hpp>
 
 using namespace std;
 
+
+template<typename T>
+void check_ser_rep(T thing, std::vector<unsigned char> expected)
+{
+    CDataStream ss(SER_DISK, 0);
+    ss << thing;
+
+    BOOST_CHECK(GetSerializeSize(thing, 0, 0) == ss.size());
+
+    std::vector<unsigned char> serialized_representation(ss.begin(), ss.end());
+
+    BOOST_CHECK(serialized_representation == expected);
+
+    T thing_deserialized;
+    ss >> thing_deserialized;
+
+    BOOST_CHECK(thing_deserialized == thing);
```

Comment:
I went down the rabbit hole trying to find out what would happen if `thing` didn't define `operator==`. To save others wondering the same thing some time, C++ is duck-typed so it will generate a compile error if you call this with a `thing` that doesn't have `operator==`. (ref http://stackoverflow.com/a/8838833).